<a href="https://colab.research.google.com/github/ahmadkiyani2452/flyrank-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## My Lane (or Freestyle) and Why

I am choosing the **Refresh / Content Opportunity Scoring** lane.

My goal is to identify which content pages should be reviewed first for refresh or optimization. Since SEO teams manage thousands of pages, it is difficult to manually decide which pages need attention. Machine learning can help prioritize pages by analyzing historical search performance and generating a ranked review queue. This supports better decision-making and helps teams focus on pages with the highest potential impact.

## The Question: Decision, Action, Cost of a Wrong Call

### Research Question

Which content pages should be reviewed first for refresh based on historical search performance?

### Decision

The project helps SEO teams decide which pages should be prioritized for review.

### Action

Pages predicted as high-priority can be refreshed, updated, expanded, or monitored before their performance declines further.

### Unit of Analysis

One content page.

### Output

A ranked list of pages with a priority score for review.

### Cost of a Wrong Recommendation

If an important page is missed, valuable search traffic may decline. If an unnecessary page is selected, the team wastes time and resources reviewing content that does not need improvement.

### Why Machine Learning?

Machine learning can identify patterns in historical search data and help prioritize pages more effectively than manual review alone.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
!git clone https://github.com/ahmadkiyani2452/flyrank-internship.git

fatal: destination path 'flyrank-internship' already exists and is not an empty directory.


In [ ]:
import os
import glob

os.chdir("/content/flyrank-internship")

csv_files = glob.glob("**/*.csv", recursive=True)

print("CSV files found:")
for file in csv_files:
    print(file)

CSV files found:
outputs/refresh_queue_sample.csv
data/raw/content_refresh_anonymized.csv
flyrank-internship/outputs/refresh_queue_sample.csv
flyrank-internship/data/raw/content_refresh_anonymized.csv


In [ ]:
import pandas as pd

# Load dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# Basic information
print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

print("\nColumn Names:")
print(df.columns.tolist())

Rows: 30000
Columns: 44

Column Names:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


In [ ]:
# 2–3 real numbers supporting the selected lane

total_pages = len(df)
unique_clients = df["client_id"].nunique()
trend_counts = df["trend_direction"].value_counts()

print("Total pages:", total_pages)
print("Unique clients:", unique_clients)

print("\nTrend direction counts:")
print(trend_counts)

Total pages: 30000
Unique clients: 32

Trend direction counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


### What these numbers show

The starter dataset contains **30,000 content pages** from **32 unique clients**. Most pages (**16,262**) show a **downward trend**, while **5,962** are stable and **4,388** are improving.

These observed numbers suggest that many pages may require content review or refresh. A machine learning model can help rank these pages based on their priority, allowing SEO teams to focus on the pages with the greatest potential impact instead of reviewing thousands of pages manually.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What I can claim

This project identifies **observed patterns** in historical search-performance data and provides a **directional priority score** for page review. The results are intended to support SEO teams in making better decisions about which pages should be reviewed first.

### What I cannot claim

This project does **not** prove that refreshing a page will definitely improve its performance. It does not predict Google's ranking algorithm or guarantee future traffic growth. The model provides **decision support**, and final decisions should always involve human judgment.


### Self-check

* [x] Every section above is filled with markdown and supporting code.
* [x] The notebook runs from top to bottom without errors.
* [x] No client names, URLs, or private queries are included.
* [x] My claims use careful words such as observed, directional, and decision-support.
* [x] The notebook is ready to commit under `work/notebooks/`.

